### PR TITLE
Bugfix/settings

### PR DESCRIPTION
**Pull request by gilleswittenberg**
Note I had to use useCallback again here for a changing dependency of the useEffect

### CODE REVIEW

File: src/components/planning/Settings.tsx [hunk 1]
```diff
@@ -160,8 +164,9 @@ const Settings: FC = () => {
         list.excludeStadia[2](excludeStadia)
       }
     })
-  }, [settingsLists]) // eslint-disable-line react-hooks/exhaustive-deps
-  // @TODO: enable
+  }
+  const initializeStadiaStable = useCallback(initializeStadia, [settingsLists])
+  useEffect(initializeStadiaStable, [settingsLists, initializeStadiaStable])
```
Comment (Jeffrey-Zutt) [inline src/components/planning/Settings.tsx:169]: Why not?

```
useEffect(() => {
if (settingsLists === undefined) return
    settingsLists.forEach(({ name, primary_stadium: primaryStadium, secondary_stadia: secondaryStadia, exclude_stadia: excludeStadia }) => {
      const list = lists.find(({ name: n }) => n === name)
      if (list === undefined) return
      if (primaryStadium !== undefined) {
        list.primaryStadium[2](primaryStadium)
      }
      if (secondaryStadia !== undefined) {
        list.secondaryStadia[2](secondaryStadia)
      }
      if (excludeStadia !== undefined) {
        list.excludeStadia[2](excludeStadia)
      }
    })
}, [settingsList])

Wrapping a method in a useCallback and then immediatly giving it to useEffect (while not calling somewhere else) feels overly complicated.

Comment (gilleswittenberg) [reply src/components/planning/Settings.tsx:169]: The dependency `lists` is the one that's changing and triggering rerender loop

Comment (Jeffrey-Zutt) [reply src/components/planning/Settings.tsx:169]: Ah now I think I see what is going on. 
Sorry, but I still think its prone to bugs.

Please have a look at:
https://reactjs.org/docs/hooks-rules.html

```
# Only Call Hooks at the Top Level

Don’t call Hooks inside loops, conditions, or nested functions. Instead, always use Hooks at the top level of your React function. By following this rule, you ensure that Hooks are called in the same order each time a component renders. That’s what allows React to correctly preserve the state of Hooks between multiple useState and useEffect calls. (If you’re curious, we’ll explain this in depth below.)
```

I don't think defining those hooks within a list is a good idea.
Can't we define every item in the list as a seperate component? (The we wouldn't need them to 'initialize' anyhow right?)

Comment (gilleswittenberg) [reply src/components/planning/Settings.tsx:169]: Will have a deeper look into this at a later stage. Merged for now.

**Pull request merged**